### PR TITLE
Base image with R (#1008)

### PR DESCRIPTION
**Pull request by Sybrand**
Test: https://wps-pr-1068.apps.silver.devops.gov.bc.ca/hfi-calculator

- renamed our base api templates and corresponding images
- added R to docker build
- added R package 'cffdrs' in docker build
- added diagram to explain imagestream flow

### CODE REVIEW

File: openshift/wps-api-base/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+RUN apt-get -y update
+RUN apt-get -y install libgdal-dev
+
+# Install R
+RUN apt-get update --fix-missing && apt-get -y install r-base
+
+# Install cffdrs
+RUN R -e "install.packages('cffdrs')"
+
+# RUN apt-get update --fix-missing && apt-get -y install r-base-dev texlive-full
```
Comment (conbrad) [inline openshift/wps-api-base/docker/Dockerfile:32]: Commented out commands

File: api/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,13 @@ RUN apt-get -y install unixodbc-dev
 # Install old (2.4.*; current debian) version of gdal
 RUN apt-get -y install libgdal-dev
 
-# Update pip  
+# Install R
```
Comment (conbrad) [inline api/Dockerfile:16]: Is this docker file intended to just be build and used locally or is it part of the openshift image build pipeline? If the latter are we installed R twice? It also gets installed in openshift/wps-api-base/docker/Dockerfile

**Review by conbrad — APPROVED** [review]

File: api/Dockerfile (hunk 2 above)
Comment (Sybrand) [reply api/Dockerfile:16]: You'll see the same command in 3 docker files - but for any one image, it's only installed once.
1) api/Dockerfile - this docker file is for local development only
2) openshift/ubuntu-base/docker/Dockerfile - this docker file only used for the c-haines cronjob (needs ubuntu for most recent gdal)
3) openshift/wps-api-base/docker/Dockerfile - this is the base image that all our PR's use. It's the fastapi docker image with our magic juice added on top.

File: api/.dockerignore [hunk 3]
```diff
@@ -7,4 +7,9 @@ grib
 .vscode
 **/__pycache__
 **/*.pyc
-python_cache
\ No newline at end of file
+python_cache
+junk/**/*
+junk
```
Comment (andrea-williams) [inline api/.dockerignore:12]: Do we want to include `junk` in this list? I presume it's a Sy-specific local folder?

File: .dockerignore [hunk 4]
```diff
@@ -1,3 +1,6 @@
 **/.gitignore
 **/node_modules
-**/Dockerfile*
\ No newline at end of file
+**/Dockerfile*
+**/openshift
+**/junk
```
Comment (andrea-williams) [inline .dockerignore:5]: Do we want to include junk in this list? I presume it's a Sy-specific local folder?

File: openshift/ubuntu-base/README.md [hunk 5]
```diff
@@ -1,29 +1,29 @@
-# Ubuntu base image.
+# Ubuntu base image
 
 This image is currently only needed for the c-haines cronjob, as it uses a more up to date version of gdal
 that is available in the api.
```
Comment (andrea-williams) [inline openshift/ubuntu-base/README.md:4]:
```suggestion
than is available in the api.
```
Should "that" -> "than" ?

**Review by andrea-williams — APPROVED** [review]

File: api/.dockerignore (hunk 3 above)
Comment (Sybrand) [reply api/.dockerignore:12]: It is Sy specific. I strongly advocate that we ignore my junk folder.

File: .dockerignore (hunk 4 above)
Comment (Sybrand) [reply .dockerignore:5]: I'm happy to rename my junk folder to something that seems less Sy specific - but I do need a place to put the many gigs of data I seem to accumulate.

**Pull request merged**